src/models/document/document-content.ts

```diff
@@ -667,11 +667,16 @@ export const DocumentContentModel = types
           if (row?.tiles.length === rowTiles.length) {
             if ((rowDropLocation === "left") || (rowDropLocation === "right")) {
               // entire row is being merged with an existing row
-              self.mergeRow(row, rowInfo);
+              const dstRow = rowInfo.rowDropIndex ? self.getRowByIndex(rowInfo.rowDropIndex) : undefined;
+              if ((rowIndex !== rowInfo.rowDropIndex) && (dstRow && !dstRow.isSectionHeader)) {
+                self.mergeRow(row, rowInfo);
+              }
             }
             else {
               // entire row is being moved to a new row
-              self.moveRowToIndex(rowIndex, rowInsertIndex);
+              if ((rowInsertIndex < rowIndex) || (rowInsertIndex > rowIndex + 1)) {
```

Comment:
At one point I thought moving a row to itself might also result in a bug. It turns out it's just inefficient, but I figured I'd leave the efficiency gain in place.